rollingpin/args.py

```diff
@@ -21,12 +32,17 @@ def __call__(self, parser, namespace, values, option_string):
 
 def _add_selection_arguments(config, parser):
     selection_group = parser.add_argument_group("host selection")
+
+    default_hosts = config["deploy"].get("default-hosts", [])
+    if not isinstance(default_hosts, list):
+        default_hosts = [default_hosts]
+
     selection_group.add_argument(
         "-h",
         action=ExtendList,
         nargs="+",
-        default=[],
-        required=True,
+        default=default_hosts,
+        required=len(default_hosts) == 0,
```

Comment:
This allows for the `hosts` argument to become optional if a default is supplied (ideally the library would be smart about this).